src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidUserDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserDataException("Користувач не може бути null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserDataException(
+                    "Користувач з таким логіном вже існує");
+        }
```

Comment:
Good job on checking if the user already exists in the storage. However, it's not mentioned in the task description, so it might be better to remove this check to adhere strictly to the task requirements.